src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -1,175 +1,506 @@
+            //
+            // Sixth, we optimize for the case where we will be comparing a single string
+            // against many potential match strings.  We cache the information about the 
+            // single string (such as it's lower-case char array) so that we don't have to
+            // recompute it.
+            //
+            // With all these changes we see an improvement of about 20x when computing the
+            // edit distance for strings.  In practical terms, what was 4 seconds of searching
+            // for results becomes around 0.2 seconds.
+
+            // First:
+            // Determine the common prefix/suffix portions of the strings.
+            while (shortLength > 0 && shortString[shortLength - 1] == longString[longLength - 1])
+            {
+                shortLength--;
+                longLength--;
+            }
 
-            // This stores the row before the current one
-            int[] rowBefore1 = GetArray(rowWidth);
+            var startIndex = 0;
+            while (startIndex < shortLength && shortString[startIndex] == longString[startIndex])
+            {
+                startIndex++;
+                shortLength--;
+                longLength--;
+            }
 
-            // This stores the current row
-            int[] rowCurrent = GetArray(rowWidth);
+            // 'shortLength' and 'longLength' are now the lengths of the substrings of our strings that we
+            // want to compare. 'startIndex' is the starting point of the substrings in both array.
 
-            try
+            // If we've matched all of the 'short' string in the prefix and suffix of 'longString'. then the edit
+            // distance is just whatever operations we have to create the remaining longString substring.
+            if (shortLength == 0)
             {
-                // Initialize the first row, which represents deleting the entire string
-                for (int column = 1; column < rowWidth; column++)
-                {
-                    rowCurrent[column] = rowCurrent[column - 1] + Delete;
-                }
+                return longLength;
+            }
 
-                for (int row = 1; row < columnHeight; row++)
-                {
-                    // Shift the row data upwards, rowBefore2 falls off and the memory is
-                    // recycled for the current row
-                    var temp = rowBefore2;
-                    rowBefore2 = rowBefore1;
-                    rowBefore1 = rowCurrent;
-                    rowCurrent = temp;
+#if false
+            // Note: this check is not necessary. Recall that shortLength is always less than 
+            // longLength.  So if longLength was 0 then shortLength would also be zero, and the
+            // above case would hit.
+            //
+            // I'm keeping this in just to help clarify this in case someone wonders why that
+            // check is missing.
+            if (longLength == 0)
```

Comment:
I'm usually pro-comment, but this explanation seems unnecessary to me. Maybe just incorporate a shortened explanation into the comment above the `if (shortLength == 0)` above.